src/Neuron/Plugin/Plugins/Links.hs

```diff
@@ -327,15 +328,20 @@ wikiLinkSpec =
             -- Cf link: [[...]]
             cmAutoLink OrdinaryConnection <$> P.try (wikiLinkP 2)
           ]
-    wikiLinkP :: Monad m => Int -> P.ParsecT [CM.Tok] s m Text
+    wikiLinkP :: Monad m => Int -> P.ParsecT [CM.Tok] s m (Text, Maybe Text)
     wikiLinkP n = do
       void $ M.count n $ symbol '['
-      s <- fmap CM.untokenize $ some $ noneOfToks [CM.Symbol ']', CM.Symbol '[', CM.LineEnd]
+      toks <- some $ noneOfToks [CM.Symbol ']', CM.Symbol '[', CM.LineEnd]
       void $ M.count n $ symbol ']'
-      pure s
-    cmAutoLink :: CM.IsInline a => Connection -> Text -> a
-    cmAutoLink conn url =
-      CM.link url title $ CM.str url
+      let parts = CM.untokenize <$> wordsBy (\x -> CM.tokType x == CM.Symbol '|') toks
+      case parts of
+        [url] -> pure (url, Nothing)
+        [url, name] -> pure (url, Just name)
+        [] -> fail "Empty wiki-link encountered"
+        _ -> fail ("Multiple pipe ('|') characters encountered; here are the parts: " ++ T.unpack (T.intercalate ", " parts))
+    cmAutoLink :: CM.IsInline a => Connection -> (Text, Maybe Text) -> a
+    cmAutoLink conn (url, name) =
+      CM.link url title $ CM.str (url `fromMaybe` name)
```

Comment:
The operator use of `fromMaybe` here is unorthodox. For future reference, I'd just write `fromMaybe url name`.